app/lib/common/widgets/room/brief_room_list_entry.dart

```diff
@@ -35,20 +36,25 @@ class BriefRoomEntry extends ConsumerWidget {
     );
     final roomName =
         ref.watch(roomDisplayNameProvider(roomId)).valueOrNull ?? roomId;
-    Widget? trailing;
-    if (trailingBuilder != null) {
-      trailing = trailingBuilder!(canLink);
-    } else if (selectedValue == roomId) {
+    Widget? trailing = trailingBuilder.let((cb) => cb(canLink));
+    if (trailing == null && selectedValue == roomId) {
       trailing = const Icon(Icons.check_circle_outline);
     }
     return ListTile(
       key: Key('$keyPrefix-$roomId'),
       enabled: canLink,
-      leading: RoomAvatarBuilder(roomId: roomId, avatarSize: 24),
+      leading: RoomAvatarBuilder(
+        roomId: roomId,
+        avatarSize: 24,
+      ),
       title: Text(roomName),
       subtitle: subtitle,
       trailing: trailing,
-      onTap: canLink && onSelect != null ? () => onSelect!(roomId) : null,
+      onTap: onSelect.let(
+        (cb) => () {
+          if (canLink) cb(roomId);
+        },
+      ),
```

Comment:
another case where if we can't link, there should not be any `onTap` at all...

```suggestion
      onTap: (canLink) ? onSelect.let(
        (cb) => () => cb(roomId),
      ) : null,
```

Reply:
Fixed by https://github.com/acterglobal/a3/pull/2241/commits/e68726840dbd837d78a9f9b1ee4285a28e4ba20c